File: mcspring-build/mcspring-plugin-manager/src/main/java/in/kyle/mcspring/manager/controller/LogFileController.java
package in.kyle.mcspring.manager.controller;

import org.springframework.stereotype.Controller;

import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

import javax.annotation.PostConstruct;

import in.kyle.mcspring.RequiresSpigot;
import lombok.SneakyThrows;

@Controller
@RequiresSpigot
class LogFileController {
    
    private final Path logsFolder = Paths.get("logs");
    
    @PostConstruct
    @SneakyThrows
    void setup() {
        Files.list(logsFolder)
                .filter(p -> p.toString().endsWith(".log.gz"))
                .forEach(this::delete);
    }
    
    @SneakyThrows
    private void delete(Path f) {
        Files.delete(f);
    }
}

File: mcspring-examples/mcspring-plugin-manager/src/main/kotlin/in/kyle/mcspring/manager/controller/LogFileController.kt
package `in`.kyle.mcspring.manager.controller

import org.bukkit.plugin.Plugin
import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
import org.springframework.stereotype.Controller
import java.nio.file.Files
import java.nio.file.Paths
import javax.annotation.PostConstruct

@Controller
internal class LogFileController {

    private val logsFolder = Paths.get("logs")

    @PostConstruct
    fun setup() {
        Files.list(logsFolder)
                .filter { it.toString().endsWith(".log.gz") }
                .forEach { Files.delete(it) }
    }
}
